envoy/stats/stats.h

```diff
@@ -92,6 +92,11 @@ class Metric : public RefcountInterface {
    */
   virtual bool used() const PURE;
 
+  /**
+   * Indicates whether this metric has it's hidden flag on.
```

Comment:
Please rephrase as to not mention the flag. That's an implementation detail and we're explicitly ignoring the flag value for counters, text readouts and histograms.